packages/experiments/src/components/Microfeedback/Microfeedback.base.tsx

```diff
@@ -0,0 +1,145 @@
+/** @jsx withSlots */
+import * as React from 'react';
+import { withSlots } from '../../Foundation';
+import { Stack } from '../../Stack';
+import { classNamesFunction } from '../../Utilities';
+import { IconButton, IButtonStyles, DefaultButton } from 'office-ui-fabric-react/lib/Button';
+import { IStackStyles } from 'office-ui-fabric-react/lib/Stack';
+import { Callout } from 'office-ui-fabric-react/lib/Callout';
+import { FocusZone, FocusZoneDirection } from 'office-ui-fabric-react/lib/FocusZone';
+import { List } from 'office-ui-fabric-react/lib/List';
+import { Text } from 'office-ui-fabric-react/lib/Text';
+import { IMicrofeedbackProps, IMicrofeedbackStyleProps, IMicrofeedbackStyles, VoteType } from './Microfeedback.types';
+import { IMicrofeedbackState } from './Microfeedback.state';
+import { initializeIcons } from '@uifabric/icons';
+
+const getClassNames = classNamesFunction<IMicrofeedbackStyleProps, IMicrofeedbackStyles>();
+
+const microfeedbackStyles: IStackStyles = {
+  root: [
+    {
+      margin: 8,
+      float: 'right'
+    }
+  ]
+};
+
+const microfeedbackItemStyles: IButtonStyles = {
+  root: [
+    {
+      border: '0px',
+      width: '100%'
+    }
+  ]
+};
+
+export class MicrofeedbackBase extends React.Component<IMicrofeedbackProps, IMicrofeedbackState> {
+  // ref's will be linked to each of the icons for callout placement
+  private dislikeRef = React.createRef<HTMLDivElement>();
+  private likeRef = React.createRef<HTMLDivElement>();
+
+  constructor(props: IMicrofeedbackProps) {
+    super(props);
+
+    initializeIcons();
+
+    this.state = {
+      // initial state of icons is neutral and followup is not visible
+      vote: 'no_vote',
+      isFollowupVisible: false
+    };
+  }
+
+  public render() {
+    const likeIcon = this.state.vote === 'like' ? 'LikeSolid' : 'Like';
+    const dislikeIcon = this.state.vote === 'dislike' ? 'DislikeSolid' : 'Dislike';
+    const hideThumbsDownCallout = this.state.vote !== 'dislike' || !this.state.isFollowupVisible;
+    const hideThumbsUpCallout = this.state.vote !== 'like' || !this.state.isFollowupVisible;
+    const onCalloutDismiss = this._onCalloutDismiss.bind(this);
+
+    const classNames = getClassNames(this.props.styles, {
+      theme: this.props.theme
+    });
+
+    return (
+      <div className={classNames.root}>
```

Comment:
I think this `div` is redundant with the `Stack` just below it. `Stack` here can be the `root`. I think we should also review this component for styles because I think it has more than just a `root` section, potentially.

Reply:
Fixed